fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeV2Literal.java

```diff
@@ -48,9 +51,30 @@ public DateTimeV2Literal(long year, long month, long day, long hour, long minute
         super(DateTimeV2Type.SYSTEM_DEFAULT, year, month, day, hour, minute, second, microSecond);
     }
 
-    public DateTimeV2Literal(DateTimeV2Type dataType,
+    public DateTimeV2Literal(DateTimeV2Type dateType,
             long year, long month, long day, long hour, long minute, long second, long microSecond) {
-        super(dataType, year, month, day, hour, minute, second, microSecond);
+        super(dateType, year, month, day, hour, minute, second, microSecond);
+        roundMicroSecond(dateType.getScale());
+    }
+
+    private void roundMicroSecond(int scale) {
+        Preconditions.checkArgument(scale >= 0 && scale <= DateTimeV2Type.MAX_SCALE,
+                "invalid datetime v2 scale: " + scale);
```

Comment:
it is better to use msg format template
```suggestion
        Preconditions.checkArgument(scale >= 0 && scale <= DateTimeV2Type.MAX_SCALE,
                "invalid datetime v2 scale: {}", scale);
```